cmd/sqlcmd/sqlcmd.go

```diff
@@ -262,66 +286,130 @@ func setFlags(rootCmd *cobra.Command, args *SQLCmdArguments) {
 	rootCmd.Flags().BoolVarP(&args.DedicatedAdminConnection, "dedicated-admin-connection", "A", false, localizer.Sprintf("Dedicated administrator connection"))
 }
 
-func normalizeFlags(rootCmd *cobra.Command) error {
+func setScriptVariable(v string) string {
+	return localizer.Sprintf("Sets the sqlcmd scripting variable %s", v)
+}
+func normalizeFlags(cmd *cobra.Command) error {
 	//Adding a validator for checking the enum flags
 	var err error
-	rootCmd.Flags().SetNormalizeFunc(func(f *pflag.FlagSet, name string) pflag.NormalizedName {
+	cmd.Flags().SetNormalizeFunc(func(f *pflag.FlagSet, name string) pflag.NormalizedName {
+		v := getFlagValueByName(f, name)
+		if v == "" {
+			return pflag.NormalizedName("")
+		}
 		switch name {
-		case "application-intent":
-			value := strings.ToLower(getFlagValueByName(f, name))
+		case applicationIntent:
+			value := strings.ToLower(v)
 			switch value {
-			case "default", "readonly":
+			case "readonly":
 				return pflag.NormalizedName(name)
 			default:
-				err = localizer.Errorf("--application-intent must be one of %s but got \"%s\"", localizer.AppIntentValues, value)
+				err = invalidParameterError(localizer.ApplicationIntentFlagShort, v, "readonly")
 				return pflag.NormalizedName("")
 			}
-		case "encrypt-connection":
-			value := strings.ToLower(getFlagValueByName(f, name))
+		case encryptConnection:
+			value := strings.ToLower(v)
 			switch value {
-			case "default", "false", "true", "disable":
+			case "false", "true", "disable":
 				return pflag.NormalizedName(name)
 			default:
-				err = localizer.Errorf("--encrypt-connection must be one of %s but got \"%s\"", localizer.EncryptConnValues, value)
+				err = invalidParameterError("-N", v, "false", "true", "disable")
 				return pflag.NormalizedName("")
 			}
-		case "format":
-			value := strings.ToLower(getFlagValueByName(f, name))
+		case format:
+			value := strings.ToLower(v)
 			switch value {
 			case "horiz", "horizontal", "vert", "vertical":
 				return pflag.NormalizedName(name)
 			default:
-				err = fmt.Errorf(localizer.Sprintf("--format must be one of %s but got \"%s\"", localizer.FormatValues, value))
+				err = invalidParameterError("-F", v, "horiz", "horizontal", "vert", "vertical")
 				return pflag.NormalizedName("")
 			}
-		case "errors-to-stderr":
-			value := getFlagValueByName(f, name)
-			switch value {
-			case "-1", "0", "1":
+		case errorsToStderr:
+			switch v {
+			case "0", "1":
 				return pflag.NormalizedName(name)
 			default:
-				err = fmt.Errorf(localizer.Sprintf("--errors-to-stderr must be one of %s but got \"%s\"", localizer.ErrToStderrValues, value))
+				err = invalidParameterError("-r", v, "0", "1")
 				return pflag.NormalizedName("")
 			}
 		}
 		return pflag.NormalizedName(name)
 	})
+	if err != nil {
+		cmd.PrintErrln(sqlcmdErrorPrefix, err.Error())
+		cmd.SilenceErrors = true
+	}
 	return err
 }
+
+var invalidArgRegexp = regexp.MustCompile(`invalid argument \"(.*)\" for \"(-.), (--.*)\"`)
```

Comment:
I was testing to see if these strings should be localizer-izsed... but loc seems to be broken in this branch:

set SQLCMD_LANG=de-de
.\sqlcmd.exe -w 4

Is returning English:

Sqlcmd: '-w 4': value must be greater than 8 and less than 65536.

It should be:

sqlcmd: Error: '-w 4': Der Wert muss größer als 8 und kleiner als 65536 sein.

Reply:
I posted a discussion in Teams about standardizing more of the error messages.

Reply:
I had a look at the test which is failing. 
After running the `go generate` command, the catalog.go gets built and the map is generated again.
The key which the testcase is trying to look for doesn't exist in the map and hence it is failing.
I'll push another commit which fixes the testcase.